extensions/php-language-features/src/features/phpGlobalFunctions.ts

```diff
@@ -1664,134 +1664,142 @@ export const globalfunctions: IEntries = {
 	},
 	fclose: {
 		description: 'Closes an open file pointer',
-		signature: '( resource $handle ): bool'
+		signature: '( resource $stream ): bool'
+	},
+	fdatasync: {
+		description: 'Synchronizes data (but not meta-data) to the file',
+		signature: '( resource $stream ): bool'
 	},
 	feof: {
 		description: 'Tests for end-of-file on a file pointer',
-		signature: '( resource $handle ): bool'
+		signature: '( resource $stream ): bool'
 	},
 	fflush: {
 		description: 'Flushes the output to a file',
-		signature: '( resource $handle ): bool'
+		signature: '( resource $stream ): bool'
 	},
 	fgetc: {
 		description: 'Gets character from file pointer',
-		signature: '( resource $handle ): string'
+		signature: '( resource $string ): string|false'
 	},
 	fgetcsv: {
 		description: 'Gets line from file pointer and parse for CSV fields',
-		signature: '( resource $handle [, int $length = 0 [, string $delimiter = "," [, string $enclosure = \'"\' [, string $escape = "\\" ]]]]): array'
+		signature: '( resource $stream [, ?int $length = null [, string $separator = "," [, string $enclosure = \'"\' [, string $escape = "\\" ]]]]): array|false'
 	},
 	fgets: {
 		description: 'Gets line from file pointer',
-		signature: '( resource $handle [, int $length ]): string'
+		signature: '( resource $stream [, ?int $length = null ]): string|false'
 	},
 	fgetss: {
 		description: 'Gets line from file pointer and strip HTML tags',
-		signature: '( resource $handle [, int $length [, string $allowable_tags ]]): string'
+		signature: '( resource $handle [, int $length = ? [, string $allowable_tags = ? ]]): string'
 	},
 	file_exists: {
 		description: 'Checks whether a file or directory exists',
 		signature: '( string $filename ): bool'
 	},
 	file_get_contents: {
 		description: 'Reads entire file into a string',
-		signature: '( string $filename [, bool $use_include_path [, resource $context [, int $offset = 0 [, int $maxlen ]]]]): string'
+		signature: '( string $filename [, bool $use_include_path = false [, ?resource $context = null [, int $offset = 0 [, ?int $maxlen = null ]]]]): string|false'
```

Comment:
Should $maxlen be $length instead?
https://www.php.net/manual/en/function.file-get-contents.php

Reply:
I suppose so, but at least it's still accurate

> Maximum length of data read